dataland-frontend/src/components/resources/frameworkDataSearch/p2p/P2pPanel.vue

```diff
@@ -0,0 +1,332 @@
+<template>
+  <div v-if="waitingForData" class="d-center-div text-center px-7 py-4">
+    <p class="font-medium text-xl">Loading P2P Data...</p>
+    <em class="pi pi-spinner pi-spin" aria-hidden="true" style="z-index: 20; color: #e67f3f" />
+  </div>
+  <div v-if="mapOfKpiKeysToDataObjectsArrays.size > 0 && !waitingForData">
+    <DataTable tableClass="onlyHeaders">
+      <Column bodyClass="headers-bg" headerStyle="width: 30vw;" headerClass="horizontal-headers-size" header="KPIs">
+      </Column>
+      <Column
+        v-for="reportingPeriodWithDataId of listOfDataSetReportingPeriods"
+        headerClass="horizontal-headers-size"
+        headerStyle="width: 30vw;"
+        :field="reportingPeriodWithDataId.dataId"
+        :header="reportingPeriodWithDataId.reportingPeriod"
+        :key="reportingPeriodWithDataId.dataId"
+      >
+      </Column>
+    </DataTable>
+    <div v-for="(arrayOfKpiDataObject, index) in mapOfKpiKeysToDataObjectsArrays" :key="index" class="d-table-style">
```

Comment:
`arrayOfKpiDataObject` is rather a key value pair with `arrayOfKpiDataObject**s**` as value.